Multichain-proposal.md

```diff
@@ -0,0 +1,112 @@
+
+The implementation is mainly developed in golang language,with a small amount of shell and C language.Leveldb database is used for local data storage,and third-party library codes such as Ethereum source code P2P and RPC modules and golang crypto are cited.
+
+The implementation provides a series of RPC interfaces for external applications,such as bridge / router,to call in an RPC driven manner.The external application initiates a keygen/signaling request(RPC call),and then calls another RPC interface to obtain the approval list for approval.When the distributed calculation is completed,it will continue to call the RPC interface to obtain the calculation results.
+
+# Cardano Bridge Integration Design
+
+Cardano will be integrated with Multichain router V3, which was lauched on June 4th 2021. The Router can allow assets to be transferred from all the chains to Cardano.
+
+The SMPC nodes will manage a single address, which will store the cardano native tokens bridged to ethereum and will mint the wrapped tokens. The only scripts needed are related to the minting policy for those wrapped tokens. This minting policy would simply check the signature of the SMPC wallet.
+
+There will be no scripts involved on the Cardano side. The SMPC nodes will manage a single wallet, and sign the transactions. 
+The SMPC wallet will store the cardano native tokens bridged to ethereum. SMPC network will mint the wrapped tokens. The only scripts needed are related to the minting policy for those wrapped tokens bridged from other chains. This minting policy would simply check the signature of the SMPC wallet.
+
+Tokens on different chains can be bridged to Cardano, but let’s simplify it to the scenario between Ethereum and Cardano.
+
+## Bringing Ethereum tokens into Cardano
+
+### **Sending ETH native tokens to Cardano**
+
+The bridge receives tokens from Ethereum chain by minting a *wrapped* version of them in Cardano blockchain. The minting is performed by the SMPC wallet using the single-issuer policy.
+
+Flow description:
+
+- User sends **k** *tokens E* using the ETH side of the bridge.
+- SMPC nodes build and sign a Cardano transaction that **mints** k *wrapped tokens E* and sends them to the user wallet.
+
+### Sending back wrapped ETH tokens from Cardano to Ethereum
+
+The user sends their wrapped tokens to the SMPC wallet. The corresponding ETH tokens are transferred to the user wallet on the Ethereum side of the bridge. The wrapped tokens will be burnt when there is incoming transaction from Ethereum. 
+
+Flow description:
+
+- User sends **k** *wrapped E* to the SMPC wallet.
+- SMPC nodes send **k** *tokens E* to the user wallet on the ETH side.
+- k *wrapped tokens E will be burnt when there is incoming transaction.*
```

Comment:
Do we need some mechanism to handle possible rollbacks? In particular on every scenario were we need to burn wrapped tokens.

Reply:
Only after the transactions reach the secure block height, the Multichain protocol can proceed the bridging request.